src/compiler/program.ts

```diff
@@ -1680,9 +1680,12 @@ namespace ts {
         function emit(sourceFile?: SourceFile, writeFileCallback?: WriteFileCallback, cancellationToken?: CancellationToken, emitOnlyDtsFiles?: boolean, transformers?: CustomTransformers, forceDtsEmit?: boolean): EmitResult {
             const tracingData: tracing.EventData = [tracing.Phase.Emit, "emit", { path: sourceFile?.path }];
             tracing.begin(...tracingData);
-            const result = runWithCancellationToken(() => emitWorker(program, sourceFile, writeFileCallback, cancellationToken, emitOnlyDtsFiles, transformers, forceDtsEmit));
-            tracing.end(...tracingData);
-            return result;
+            try {
+                return runWithCancellationToken(() => emitWorker(program, sourceFile, writeFileCallback, cancellationToken, emitOnlyDtsFiles, transformers, forceDtsEmit));

```

Comment:
Same here.. runWithCancellationToken already has try catch finally. thats where it should be able to handle the tracing.. the data to revert back to could be parameter to runWithCancellationToken function instead